### PR TITLE
Fix safeToBetaReduce in PIR call site inliner

### PR DESCRIPTION
**Pull request by thealmarty**
Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Changelog fragments have been written (if appropriate)
    - [x] Relevant tickets are mentioned in commit messages
    - [x] Formatting, PNG optimization, etc. are updated
- PR
    - [ ] (For external contributions) Corresponding issue exists and is linked in the description
    - [x] Targeting master unless this is a cherry-pick backport
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested

### CODE REVIEW

**Comment by thealmarty** [comment]
Investigation from #5395  reveals this bug. Still investigating the failed test. It was definitely inlining `addInteger` to the `(builtin addInteger)` function so I took out that binding to isolate what it's doing. I still can't make sense of it though.

File: plutus-core/plutus-ir/src/PlutusIR/Transform/Inline/CallSiteInline.hs [hunk 1]
```diff
@@ -165,7 +165,7 @@ fullyApplyAndBetaReduce info args0 = do
         -- `arg`
         Term tyname name uni fun ann ->
         InlineM tyname name uni fun ann Bool
-      safeToBetaReduce a = shouldUnconditionallyInline Strict a rhsBody
+      safeToBetaReduce a arg = shouldUnconditionallyInline Strict a arg rhsBody
```
Comment (effectfully) [inline plutus-core/plutus-ir/src/PlutusIR/Transform/Inline/CallSiteInline.hs:168]: Oh wow.

**Review by michaelpj — APPROVED** [review]
Fix seems correct, and the tests seem improved, looks good!

File: plutus-core/plutus-ir/test/transform/inline/callsite-non-trivial-body [hunk 2]
```diff
@@ -40,11 +35,11 @@ size ((+) 6 (1 + 2 + 3 + 4)) > size (f (\unused -> 5) (+) 6)
             [ a b ]
             [
               [
-                addInteger
+                (builtin addInteger)
```
[the remark this answers is not in the document]
Comment (thealmarty) [reply plutus-core/plutus-ir/test/transform/inline/callsite-non-trivial-body:38]: This is me taking out the let binding of `let addInteger = builtin addInteger` in the test case to not confuse us on what it's inline. This is the input file.

[the remark this answers is not in the document]
Comment (michaelpj) [reply plutus-core/plutus-ir/test/transform/inline/callsite-non-trivial-body:38]: Oh lol. So this has no effect at all? That seems surprising!

File: plutus-core/plutus-ir/test/transform/inline/callsite-non-trivial-body.golden [hunk 3]
```diff
@@ -48,37 +43,29 @@
...
       [
-        [ [ f (lam unused (con integer) (con integer 5)) ] addInteger ]
+        [
+          [ f (lam unused (con integer) (con integer 5)) ] (builtin addInteger)
+        ]
         (con integer 6)
       ]
     ]
     [
+      [ (builtin addInteger) (con integer 11) ]
```
Comment (thealmarty) [inline plutus-core/plutus-ir/test/transform/inline/callsite-non-trivial-body.golden:55]: @michaelpj this is the new test result and I can't figure out why it's doing this. If you look at [the output file](https://github.com/input-output-hk/plutus/blob/3c9e8f7716dfbc848ce5e2685d56903661ec0f62/plutus-core/plutus-ir/test/transform/inline/callsite-non-trivial-body.golden) it doesn't seem right.

**Comment by thealmarty** [comment]
Some budgets got more improvements! Yay! Not sure about the `plutus-tx-plugin` tests. Can someone @input-output-hk/plutus-core check those please.

**Review by michaelpj — APPROVED** [review]

**Pull request merged**